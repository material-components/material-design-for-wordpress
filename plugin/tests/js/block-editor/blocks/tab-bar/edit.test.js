/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import '@testing-library/jest-dom/extend-expect';
import { render, fireEvent } from '@testing-library/react';

/**
 * Internal dependencies
 */
import Edit from '../../../../../assets/src/block-editor/blocks/tab-bar/edit';

jest.mock( '@wordpress/blocks', () => ( {
	getBlockTypes: () => [],
} ) );

jest.unmock( '@wordpress/data' );
jest.mock( '@wordpress/data', () => ( {
	combineReducers: jest.fn(),
	registerStore: jest.fn(),
	select: store => {
		switch ( store ) {
			case 'core/block-editor':
				return {
					getBlocks: () => [],
				};

			default:
				return {};
		}
	},
	withSelect: mapSelectToProps => WrappedComponent => ownProps => {
		return (
			<WrappedComponent
				{ ...ownProps }
				{ ...mapSelectToProps }
				tabContent={ [] }
			/>
		);
	},
	dispatch: store => {
		switch ( store ) {
			case 'core/block-editor':
				return {
					replaceInnerBlocks: jest.fn(),
					removeBlocks: jest.fn(),
				};

			default:
				return {};
		}
	},
} ) );

/**
 * Render the component.
 *
 * @param {Object} props - Component props
 * @return {Function} A functional component.
 */
const setup = props => {
	return render( <Edit { ...props } /> );
};

const tabProps = {
	attributes: {
		tabs: [
			{
				label: 'Tab 1 Title',
				content: [
					{
						clientId: 'c0cf3523-d407-4389-939d-b1d1007b316f',
						name: 'core/paragraph',
						isValid: true,
						attributes: {
							content: 'Tab 1 content',
							dropCap: false,
						},
						innerBlocks: [],
					},
				],
			},
			{
				label: 'Tab 2 Title',
				content: [
					{
						clientId: '00d62a82-e010-49cb-982a-bc04767df7ea',
						name: 'core/paragraph',
						isValid: true,
						attributes: {
							content: 'Tab 2 content',
							dropCap: false,
						},
						innerBlocks: [],
					},
				],
			},
		],
		iconPosition: 'none',
	},
	className: 'wp-block-material-tab-bar',
	setAttributes: jest.fn(),
};

describe( 'blocks: material/tab-bar: Edit', () => {
	it( 'matches snapshot', () => {
		const wrapper = setup( tabProps );
		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'selects a tab if the title is clicked', () => {
		const props = { ...tabProps };
		const { container, getByText } = setup( props );

		fireEvent.click( getByText( 'Tab 2 Title' ) );
		const tabItems = container.querySelectorAll( '.mdc-tab' );

		expect( tabItems[ 1 ].className ).toContain( 'mdc-tab--active' );
	} );
} );
