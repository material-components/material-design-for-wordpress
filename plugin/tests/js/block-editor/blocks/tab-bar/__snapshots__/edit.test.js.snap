// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blocks: material/tab-bar: Edit matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <p
      class="a11y-speak-intro-text"
      hidden="hidden"
      id="a11y-speak-intro-text"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    >
      Notifications
    </p>
    <div
      aria-atomic="true"
      aria-live="assertive"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-assertive"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div
      aria-atomic="true"
      aria-live="polite"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-polite"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div>
      <div
        class="tab-wrap"
      >
        <div
          class="mdc-tab-bar"
          role="tablist"
        >
          <div
            class="mdc-tab-scroller"
          >
            <div
              class="mdc-tab-scroller__scroll-area mdc-tab-scroller__scroll-area--scroll"
              style="margin-bottom: 0px;"
            >
              <div
                class="mdc-tab-scroller__scroll-content"
              >
                <div
                  class="mdc-tab tab mdc-tab--active"
                  id="mdc-tab-1"
                  role="tab"
                  tabindex="0"
                >
                  <span
                    class="mdc-tab__content"
                  >
                    <span
                      class="mdc-tab__text-label tab__label-field"
                    >
                      <span
                        aria-label="Tab Title"
                        aria-multiline="true"
                        class="rich-text block-editor-rich-text__editable"
                        role="textbox"
                        style="white-space: pre-wrap;"
                      >
                        Tab 1 Title
                      </span>
                    </span>
                  </span>
                  <span
                    class="mdc-tab-indicator mdc-tab-indicator--active"
                  >
                    <span
                      class="mdc-tab-indicator__content mdc-tab-indicator__content--underline"
                    />
                  </span>
                  <span
                    class="mdc-tab__ripple"
                  />
                  <button
                    class="material-icons tab__delete"
                  >
                    cancel
                  </button>
                </div>
                <div
                  class="mdc-tab tab"
                  id="mdc-tab-2"
                  role="tab"
                  tabindex="0"
                >
                  <span
                    class="mdc-tab__content"
                  >
                    <span
                      class="mdc-tab__text-label tab__label-field"
                    >
                      <span
                        aria-label="Tab Title"
                        aria-multiline="true"
                        class="rich-text block-editor-rich-text__editable"
                        role="textbox"
                        style="white-space: pre-wrap;"
                      >
                        Tab 2 Title
                      </span>
                    </span>
                  </span>
                  <span
                    class="mdc-tab-indicator"
                  >
                    <span
                      class="mdc-tab-indicator__content mdc-tab-indicator__content--underline"
                    />
                  </span>
                  <span
                    class="mdc-tab__ripple"
                  />
                  <button
                    class="material-icons tab__delete"
                  >
                    cancel
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <button
          class="tab-add"
        >
          <i
            class="material-icons tab-add__icon"
          >
            add
          </i>
          <span>
            New Tab
          </span>
        </button>
      </div>
      <div>
         
        {"id":"button-1","name":"material/button"}
         
      </div>
      <div
        class="components-toolbar"
        label="Change the tab order"
      >
        <div>
          <button
            aria-label="Move left"
            class="components-button components-toolbar__control has-icon"
            type="button"
          >
            <i
              class="material-icons"
            >
              keyboard_arrow_left
            </i>
          </button>
        </div>
        <div>
          <button
            aria-label="Move right"
            class="components-button components-toolbar__control has-icon"
            type="button"
          >
            <i
              class="material-icons"
            >
              keyboard_arrow_right
            </i>
          </button>
        </div>
      </div>
      <div
        class="components-panel__body is-opened"
      >
        <h2
          class="components-panel__body-title"
        >
          <button
            aria-expanded="true"
            class="components-button components-panel__body-toggle"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              <svg
                aria-hidden="true"
                class="components-panel__arrow"
                focusable="false"
                height="24"
                role="img"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 8l-6 5.4 1 1.2 5-4.6 5 4.6 1-1.2z"
                />
              </svg>
            </span>
            Icon
          </button>
        </h2>
        <div
          class="btn-group"
        >
          <ul
            class="btn-group__list"
          >
            <li
              class="btn-group__list__list-item"
            >
              <button
                class="btn-group__list__list-item__button btn-group__list__list-item__button--active"
                type="button"
              >
                None
              </button>
            </li>
            <li
              class="btn-group__list__list-item"
            >
              <button
                class="btn-group__list__list-item__button"
                type="button"
              >
                Leading
              </button>
            </li>
            <li
              class="btn-group__list__list-item"
            >
              <button
                class="btn-group__list__list-item__button"
                type="button"
              >
                Above
              </button>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="tab-wrap"
    >
      <div
        class="mdc-tab-bar"
        role="tablist"
      >
        <div
          class="mdc-tab-scroller"
        >
          <div
            class="mdc-tab-scroller__scroll-area mdc-tab-scroller__scroll-area--scroll"
            style="margin-bottom: 0px;"
          >
            <div
              class="mdc-tab-scroller__scroll-content"
            >
              <div
                class="mdc-tab tab mdc-tab--active"
                id="mdc-tab-1"
                role="tab"
                tabindex="0"
              >
                <span
                  class="mdc-tab__content"
                >
                  <span
                    class="mdc-tab__text-label tab__label-field"
                  >
                    <span
                      aria-label="Tab Title"
                      aria-multiline="true"
                      class="rich-text block-editor-rich-text__editable"
                      role="textbox"
                      style="white-space: pre-wrap;"
                    >
                      Tab 1 Title
                    </span>
                  </span>
                </span>
                <span
                  class="mdc-tab-indicator mdc-tab-indicator--active"
                >
                  <span
                    class="mdc-tab-indicator__content mdc-tab-indicator__content--underline"
                  />
                </span>
                <span
                  class="mdc-tab__ripple"
                />
                <button
                  class="material-icons tab__delete"
                >
                  cancel
                </button>
              </div>
              <div
                class="mdc-tab tab"
                id="mdc-tab-2"
                role="tab"
                tabindex="0"
              >
                <span
                  class="mdc-tab__content"
                >
                  <span
                    class="mdc-tab__text-label tab__label-field"
                  >
                    <span
                      aria-label="Tab Title"
                      aria-multiline="true"
                      class="rich-text block-editor-rich-text__editable"
                      role="textbox"
                      style="white-space: pre-wrap;"
                    >
                      Tab 2 Title
                    </span>
                  </span>
                </span>
                <span
                  class="mdc-tab-indicator"
                >
                  <span
                    class="mdc-tab-indicator__content mdc-tab-indicator__content--underline"
                  />
                </span>
                <span
                  class="mdc-tab__ripple"
                />
                <button
                  class="material-icons tab__delete"
                >
                  cancel
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        class="tab-add"
      >
        <i
          class="material-icons tab-add__icon"
        >
          add
        </i>
        <span>
          New Tab
        </span>
      </button>
    </div>
    <div>
       
      {"id":"button-1","name":"material/button"}
       
    </div>
    <div
      class="components-toolbar"
      label="Change the tab order"
    >
      <div>
        <button
          aria-label="Move left"
          class="components-button components-toolbar__control has-icon"
          type="button"
        >
          <i
            class="material-icons"
          >
            keyboard_arrow_left
          </i>
        </button>
      </div>
      <div>
        <button
          aria-label="Move right"
          class="components-button components-toolbar__control has-icon"
          type="button"
        >
          <i
            class="material-icons"
          >
            keyboard_arrow_right
          </i>
        </button>
      </div>
    </div>
    <div
      class="components-panel__body is-opened"
    >
      <h2
        class="components-panel__body-title"
      >
        <button
          aria-expanded="true"
          class="components-button components-panel__body-toggle"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            <svg
              aria-hidden="true"
              class="components-panel__arrow"
              focusable="false"
              height="24"
              role="img"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 8l-6 5.4 1 1.2 5-4.6 5 4.6 1-1.2z"
              />
            </svg>
          </span>
          Icon
        </button>
      </h2>
      <div
        class="btn-group"
      >
        <ul
          class="btn-group__list"
        >
          <li
            class="btn-group__list__list-item"
          >
            <button
              class="btn-group__list__list-item__button btn-group__list__list-item__button--active"
              type="button"
            >
              None
            </button>
          </li>
          <li
            class="btn-group__list__list-item"
          >
            <button
              class="btn-group__list__list-item__button"
              type="button"
            >
              Leading
            </button>
          </li>
          <li
            class="btn-group__list__list-item"
          >
            <button
              class="btn-group__list__list-item__button"
              type="button"
            >
              Above
            </button>
          </li>
        </ul>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
